src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -31,258 +31,281 @@
 #include "application_manager/event_engine/event_observer.h"
 #include "application_manager/message_helper.h"
 #include "application_manager/resumption/resumption_data_processor.h"
 #include "utils/helpers.h"
+#include "vehicle_info_plugin/custom_vehicle_data_manager.h"
 
 namespace vehicle_info_plugin {
+CREATE_LOGGERPTR_GLOBAL(logger_, "VehicleInfoPlugin")
 
-CREATE_LOGGERPTR_GLOBAL(logger_, "VehicleInfoPendingResumptionHandler")
+using hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData;
 
-VehicleInfoPendingResumptionHandler::VehicleInfoPendingResumptionHandler(
-    application_manager::ApplicationManager& application_manager)
-    : ExtensionPendingResumptionHandler(application_manager) {}
-
-template <class Key, class Value>
-std::set<Key> EnumerateKeys(std::map<Key, Value>& container) {
-  std::set<std::string> keys;
-
-  std::transform(
-      container.begin(),
-      container.end(),
-      std::inserter(keys, keys.end()),
-      [&](const std::pair<std::string, bool>& pair) { return pair.first; });
+uint32_t get_corr_id_from_message(const smart_objects::SmartObject& message) {
+  using namespace application_manager;
+  return message[strings::params][strings::correlation_id].asInt();
+}
 
-  return keys;
+template <class T>
+std::string Stringify(const T& container) {
+  std::stringstream ss;
+  for (const auto& val : container) {
+    ss << val << " ";
+  }
+  return ss.str();
 }
 
-bool VehicleInfoPendingResumptionHandler::IsResumptionResultSuccessful(
-    std::map<std::string, bool>& subscription_results) {
-  for (auto ivi_status : subscription_results) {
-    if (!ivi_status.second) {
-      return false;
-      break;
+std::set<std::string> SubscriptionsFromResponse(
+    const smart_objects::SmartObject& response,
+    std::function<bool(const smart_objects::SmartObject& vehicle_data)>
+        vehicle_data_check) {
+  namespace strings = application_manager::strings;
+  std::set<std::string> result;
+  const auto response_params = response[strings::msg_params];
+  const auto response_keys = response_params.enumerate();
+  for (auto key : response_keys) {
+    if (vehicle_data_check(response_params[key])) {
+      result.insert(key);
     }
   }
+  return result;
+}
 
-  return true;
+bool IsResponseSuccessful(const smart_objects::SmartObject& response) {
+  namespace strings = application_manager::strings;
+  if (!response[strings::params].keyExists(strings::error_msg)) {
+    return true;
+  }
+  return false;
 }
 
-void VehicleInfoPendingResumptionHandler::RemoveSucessfulSubscriptions(
-    std::set<std::string>& subscriptions,
-    std::set<std::string>& successful_subscriptions) {
-  for (auto subscription : subscriptions) {
-    if (helpers::in_range(successful_subscriptions, subscription)) {
-      subscriptions.erase(subscription);
-    }
+void FillResponseWithMissedVD(const std::set<std::string>& vehicle_data,
+                              smart_objects::SmartObject* response) {
+  DCHECK(response)
```

Comment:
```suggestion
  DCHECK_OR_RETURN_VOID(response)
```

Reply:
I would say that if FillResponseWithMissedVD will be alled with nullptr in release we need to crash. Other way business logic will be wrong and we will have a lot of wonderful time to find out what was the issue